### PR TITLE
Task Id

### PR DESCRIPTION
**Pull request by KellyLi**
Adding task id (time of task creation in ms), still need to write tests

### CODE REVIEW

File: src/main/java/seedu/task/model/util/SampleDataUtil.java [hunk 1]
```diff
@@ -12,12 +12,12 @@
     public static Task[] getSampleTasks() {
         try {
             return new Task[] {
-                new Task(new Description("Alex Yeoh"), null, null, new UniqueTagList("friends")),
-                new Task(new Description("Bernice Yu"), null, null, new UniqueTagList("colleagues", "friends")),
-                new Task(new Description("Charlotte Oliveiro"), null, null, new UniqueTagList("neighbours")),
-                new Task(new Description("David Li"), null, null, new UniqueTagList("family")),
-                new Task(new Description("Irfan Ibrahim"), null, null, new UniqueTagList("classmates")),
-                new Task(new Description("Roy Balakrishnan"), null, null, new UniqueTagList("colleagues"))
+                new Task(new Description("Alex Yeoh"), null, null, new UniqueTagList("friends"), null),
```
Comment (evanyeung) [inline src/main/java/seedu/task/model/util/SampleDataUtil.java:15]: It looks like you've made the `id` field non-nullable in the constructor (which seems right to me). Will this cause tests to fail?

File: src/test/java/seedu/task/testutil/TestUtil.java [hunk 2]
```diff
@@ -72,15 +72,15 @@ public static void assertThrows(Class<? extends Throwable> expected, Runnable ex
...
-                new Task(new Description("Ali Muster"), null, null, new UniqueTagList()),
-                new Task(new Description("Boris Mueller"), null, null, new UniqueTagList()),
-                new Task(new Description("Carl Kurz"), null, null, new UniqueTagList()),
-                new Task(new Description("Daniel Meier"), null, null, new UniqueTagList()),
-                new Task(new Description("Elle Meyer"), null, null, new UniqueTagList()),
-                new Task(new Description("Fiona Kunz"), null, null, new UniqueTagList()),
-                new Task(new Description("George Best"), null, null, new UniqueTagList()),
-                new Task(new Description("Hoon Meier"), null, null, new UniqueTagList()),
-                new Task(new Description("Ida Mueller"), null, null, new UniqueTagList())
+                new Task(new Description("Ali Muster"), null, null, new UniqueTagList(), null),
```
Comment (evanyeung) [inline src/test/java/seedu/task/testutil/TestUtil.java:75]: Same as above about asserting null id in the constructor.

File: src/main/java/seedu/task/model/task/Task.java [hunk 3]
```diff
@@ -11,7 +11,7 @@
  */
 public class Task implements ReadOnlyTask {
 
-    private TaskId id;
+    private final TaskId id;
```
Comment (evanyeung) [inline src/main/java/seedu/task/model/task/Task.java:14]: Nice, thanks!

**Review by evanyeung — CHANGES_REQUESTED** [review]
Looks good! Marking that I looked at it but requesting changes till tests are fixed.

**Comment by KellyLi** [comment]
Closing and reopening on another branch with fixed tests

**Pull request closed without merging**